core/controllers/admin_test.py

```diff
@@ -128,12 +128,12 @@ def test_change_rights(self):
         self.signup(
             BOTH_MODERATOR_AND_ADMIN_EMAIL, BOTH_MODERATOR_AND_ADMIN_USERNAME)
 
-        # Navigate to any page. The role is not set.
-        self.testapp.get('/').mustcontain(no=['/moderator', '/admin'])
+        # There should be a redirect on navigating to /
+        self.assertEqual(self.testapp.get('/').status_int, 302)
```

Comment:
Actually, with the new structure, you probably don't need these two lines here. I recommend adding another test altogether (in base_test.py) that checks that the redirection happens correctly in both the logged-out and logged-in cases, and that the destination pages are correct.